sources/osgShader/Compiler.js

```diff
@@ -627,21 +629,14 @@ define( [
 
             var samplerName = 'Texture' + unit.toString();
             var textureSampler = this.getVariable( samplerName );
-
             if ( textureSampler === undefined ) {
-
-                if ( texture.className() === 'Texture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                } else if ( texture.className() === 'TextureCubeMap' ) {
-                    textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
-                } else if ( texture.className() === 'ShadowTexture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                    // return now to prevent creation of useless FragTexCoord
-                    //( shadow creates its own texcoord)
-                    return;
-                }
-
-
+                textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
+            } else if ( texture.className() === 'TextureCubeMap' ) {
+                textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
+            } else if ( texture.className() === 'ShadowTexture' ) {
```

Comment:
it's part of the thing that need refactor/thinking:
Sampler2D and Int shader Texture location uniform are actually both sharing the same name thus breaking compiler uniform duplicate detection of same uniform with different types